stl/inc/memory

```diff
@@ -308,10 +308,22 @@ _NoThrowFwdIt uninitialized_fill_n(_NoThrowFwdIt _First, const _Diff _Count_raw,
 }
 #endif // _HAS_IF_CONSTEXPR
 
+#if _HAS_CXX20
+template <class _Ty, class... _Types>
+_Ty* construct_at(_Ty* const _Location, _Types&&... _Args) noexcept(
+    is_nothrow_constructible_v<_Ty, _Types...>) /* strengthened */ {
+    return ::new (const_cast<void*>(static_cast<const volatile void*>(_Location))) _Ty(_STD forward<_Types>(_Args)...);
+}
+
+namespace ranges {
+    using _STD construct_at;
+} // namespace ranges
+#endif // _HAS_CXX20
+
 #if _HAS_CXX17
 // FUNCTION TEMPLATE destroy_at
 template <class _Ty>
-void destroy_at(_Ty* const _Location) { // destroy _Ty at memory address _Location
+void destroy_at(_Ty* const _Location) noexcept(noexcept(_Location->~_Ty())) /* strengthened */ {

```

Comment:
This should be strengthened to unconditional `noexcept`. According to WG21-N4849 [res.on.functions], the STL can assume that user-provided destructors never emit exceptions:

"In certain cases (replacement functions, handler functions, operations on types used to instantiate standard library template components), the C++ standard library depends on components supplied by a C++ program. If these components do not meet their requirements, this document places no requirements on the implementation.
In particular, the effects are undefined in the following cases:
[...]
If any replacement function or handler function or destructor operation exits via an exception, unless specifically allowed in the applicable *Required behavior:* paragraph."